### ISSUE TITLE
selector on specific pod names auto capitalizes the value of the pod's namespace

### ISSUE THREAD

**Issue by jaysunmah** [chaos-mesh/chaos-mesh#1505]
## Bug Report

**What version of Kubernetes are you using?**
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-16T00:04:31Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.14-airbnb0", GitCommit:"9931e7ec32e27c7ef21b3a734d8554a5ebadf00c", GitTreeState:"clean", BuildDate:"2020-12-02T19:00:47Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**What version of Chaos Mesh are you using?**

```
Controller manager Version: version.Info{GitVersion:"v0.9.0-156+a7b83630d43439", GitCommit:"a7b83630d43439b43c091842b10ed6ed6166f69b", BuildDate:"2021-01-12T19:01:35Z", GoVersion:"go1.14.6", Compiler:"gc", Platform:"linux/amd64"}
```
**What did you do?**

When I try to apply a `NetworkChaos` yaml resource, such as:
```
---
apiVersion: chaos-mesh.org/v1alpha1
kind: NetworkChaos
metadata:
  name: test
  namespace: boom2-chaos-mesh
spec:
  direction: both
  action: partition
  mode: all
  duration: 86400s
  selector:
    namespaces:
    - uds-ppe-a
    pods:
      uds-ppe-a:
      - use1e-lookups-80-x-ebs-replica-1
  externalTargets:
  - 0.0.0.0/1
  - 128.0.0.0/2
  - 192.0.0.0/3
  - 224.0.0.0/4
  scheduler:
    cron: "@every 86401s"
```

The namespace key in `selector.pods` (in our example `uds-ppe-a`) gets auto capitalized (to `Uds - Ppe - A`), and thus the resource is erroring saying it can't find any pods in that namespace. The resulting k8s resource when applied looks like
```
...
Spec:
  Action:     partition
  Direction:  both
  Duration:   86400s
  External Targets:
    0.0.0.0/1
    128.0.0.0/2
    192.0.0.0/3
    224.0.0.0/4
  Mode:  all
  Scheduler:
    Cron:  @every 86401s
  Selector:
    Namespaces:
      uds-ppe-a
    Pods:
      Uds - Ppe - A:
        use1e-lookups-80-x-ebs-replica-1
  Value:
Status:
  Experiment:
    Phase:         Failed
  Failed Message:  no pod is selected
  Phase:
  Scheduler:
Events:
  Type     Reason             Age               From                     Message
  ----     ------             ----              ----                     -------
  Warning  ChaosInjectFailed  1s (x10 over 4s)  networkchaos-controller  no pod is selected
```
If it helps, I noticed when I change the value of the namespace to include a number, such as `uds-ppe-1`, that doesn't get auto capitalized. 

**What did you expect to see?**
The resulting spec should still try to inject chaos onto pods in namespace `uds-ppe-a`

**What did you see instead?**
The resulting spec's namespace is `Uds - Ppe - A`

**Output of chaosctl**

**Comment by cwen0** [comment]
@jaysunmah Thanks for your report.  I have tested the same YAML file on version v1.1.1, but I can't reproduce this issue. 

```
spec:
  action: partition
  direction: both
  duration: 86400s
  externalTargets:
  - 0.0.0.0/1
  - 128.0.0.0/2
  - 192.0.0.0/3
  - 224.0.0.0/4
  mode: all
  scheduler:
    cron: '@every 86401s'
  selector:
    namespaces:
    - local-path-storage
    pods:
      local-path-storage:
      - local-path-provisioner-85445b74d4-qjfct
  value: ""
status:
  experiment:
``` 

I can't find the git [commit(a7b83630d43439b43c091842b10ed6ed6166f69b)](https://github.com/chaos-mesh/chaos-mesh/search?q=a7b83630d43439b43c091842b10ed6ed6166f69b&type=commits) your reported.  Can you check which image you are using?

**Comment by jaysunmah** [comment]
hey @cwen0 , thanks for the reply. The git commit you're referencing isn't from the public chaos-mesh repo, my company (Airbnb) has a private fork of it where we manage the code version ourselves. 

I repulled from upstream just now, and I have my version now set as `Controller manager Version: version.Info{GitVersion:"master-g2898443ea734b4", GitCommit:"2898443ea734b45f5e94df1a655695d4ea439af3", BuildDate:"2021-02-18T05:34:14Z", GoVersion:"go1.14.6", Compiler:"gc", Platform:"linux/amd64"}`.

However, when I reapply the resource file, the namespace issue is still there. (expecting `uds-ppe-a` , but getting `Uds - Ppe - A` instead). Similarly, when I apply the resource YAML you provided, I'm seeing it as 
```
    Pods:
      Local - Path - Storage:
``` 
instead of 
```
    Pods:
      local-path-storage:
```

**Comment by jaysunmah** [comment]
agh, @cwen0  I actually was able to figure out my issue. My problem was that I had only enabled specific namespaces in my helm chart via `allowedNamespaces`, so the new namespace I switched to was considered invalid. 

For feedback, would it be possible to change this behavior to output a fat error immediately if someone tries injecting chaos onto an invalid namespace, instead of the current behavior which simply says "No pod is selected"?
